rust/rbac-registration/src/cardano/cip509/cip509.rs

```diff
@@ -0,0 +1,497 @@
+//! Cardano Improvement Proposal 509 (CIP-509) metadata module.
+//! Doc Reference: <https://github.com/input-output-hk/catalyst-CIPs/tree/x509-envelope-metadata/CIP-XXXX>
+//! CDDL Reference: <https://github.com/input-output-hk/catalyst-CIPs/blob/x509-envelope-metadata/CIP-XXXX/x509-envelope.cddl>
+
+use anyhow::anyhow;
+use cardano_blockchain_types::{
+    hashes::{Blake2b256Hash, BLAKE_2B256_SIZE},
+    Slot, TxnIndex,
+};
+use catalyst_types::problem_report::ProblemReport;
+use minicbor::{
+    decode::{self},
+    Decode, Decoder,
+};
+use pallas::{
+    codec::utils::Nullable,
+    ledger::traverse::{MultiEraBlock, MultiEraTx},
```

Comment:
Let's use our `MultiEraBlock` , the current one doesn't have functionality that you want, but will add one